Core/MMKV_IO.cpp

```diff
@@ -765,14 +791,27 @@ bool MMKV::removeDataForKey(MMKVKey_t key) {
     return false;
 }
 
-KVHolderRet_t
-MMKV::doAppendDataWithKey(const MMBuffer &data, const MMBuffer &keyData, bool isDataHolder, uint32_t originKeyLength) {
-    auto isKeyEncoded = (originKeyLength < keyData.length());
-    auto keyLength = static_cast<uint32_t>(keyData.length());
+uint32_t MMKV::getValueLength(const mmkv::MMBuffer &data, bool isDataHolder) {
     auto valueLength = static_cast<uint32_t>(data.length());
     if (isDataHolder) {
         valueLength += pbRawVarint32Size(valueLength);
     }
+    return valueLength;
+}
+
+void MMKV::writeValueData(const MMBuffer &data, bool isDataHolder, CodedOutputData *output) {
```

Comment:
Again, you should never put MMKV unrelative methods into the MMKV class. It's really unprofessional because MMKV is a public class. And it's a little annoying that such things have been corrected before.